common/cache/lru.go

```diff
@@ -309,8 +310,9 @@ func (c *lru) putInternal(key interface{}, value interface{}, allowUpdate bool)
 	// If the entry exists, check if it has expired or update the value
 	if elt != nil {
 		existingEntry := elt.Value.(*entryImpl)
-		if !c.isEntryExpired(existingEntry, time.Now().UTC()) {
+		if !c.isEntryExpired(existingEntry, c.timeSource.Now().UTC()) {
```

Comment:
I believe this is an oversight; it should use the timesource and not `time` directly. Without this, my test wasn't passing.